src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java

```diff
@@ -196,4 +198,10 @@ public String getReleaseAnnotation(ReleaseLevel releaseLevel) {
   public String getBatchingDescriptorConstName(Method method) {
     return inittedConstantName(Name.upperCamel(method.getSimpleName()).join("batching_desc"));
   }
+
+  @Override
+  public String getArtifactId(Model model) {
+    List<String> nameTokens = Splitter.on(".").splitToList(model.getServiceConfig().getName());
```

Comment:
Fair enough. We may want to file an issue to make this information available in toolkit, either via the command line or a gapic yaml setting.